packages/framework-core/src/booster-auth.ts

```diff
@@ -62,6 +65,62 @@ export class BoosterAuth {
     // Check the roles then
     return userHasSomeRole(user, authorizedRoles)
   }
+
+  public static async verifyToken(config: BoosterConfig, token: string): Promise<UserEnvelope> {
+    if (!config.tokenVerifier) {
+      throw new Error('Token verifier not configured')
+    }
+    const { issuer, jwksUri } = config.tokenVerifier
+
+    const client = jwksRSA({
+      jwksUri,
+    })
+
+    const verifyOptions: jwt.VerifyOptions = {
+      algorithms: ['RS256'],
+      issuer,
+    }
+
+    return new Promise((resolve, reject) => {
+      const getKey = (header: jwt.JwtHeader, callback: jwt.SigningKeyCallback): void => {
+        if (!header.kid) {
+          callback(new Error('JWT kid not found.'))
+          return
+        }
+
+        client.getSigningKey(header.kid, function(err: Error | null, key: jwksRSA.SigningKey) {
+          if (err) {
+            // This callback doesn't accept null so an empty string is enough here
+            callback(err, '')
+            return
+          }
+          const signingKey = key.getPublicKey()
+          callback(null, signingKey)
+        })
+      }
+
+      token = BoosterAuth.sanitizeToken(token)
+      jwt.verify(token, getKey, verifyOptions, (err: Error | null, decoded: object | undefined) => {
+        if (err) {
+          return reject(err)
+        }
+        return resolve(BoosterAuth.tokenToUserEnvelope(decoded))
+      })
+    })
+  }
+
+  private static tokenToUserEnvelope(decodedToken: any): UserEnvelope {
+    const username = decodedToken.email ?? decodedToken.phone_number
+    const role = decodedToken['custom:role']
+    return {
+      username,
+      role: role?.trim() ?? '',
+    }
```

Comment:
@charlietfe I forgot! Could you please add here (and to the UserEnvelope type) a field called `id` that contains the id of the user? (it is the `sub` claim)

We have received several requests for having that

Reply:
Sure thing!

Reply:
Thanks!